lib/util/index.js

```diff
@@ -104,25 +104,68 @@ function flatMap(list, mapFn) {
   }, []);
 }
 
+/**
+ * Get the browser's user agent, if available.
+ * @returns {?string}
+ */
+function getUserAgent() {
+  return typeof navigator !== 'undefined' && typeof navigator.userAgent === 'string'
+    ? navigator.userAgent
+    : null;
+}
+
 /**
  * Guess the browser.
+ * @param {string} [userAgent=navigator.userAgent]
  * @returns {?string} browser - "chrome", "firefox", "safari", or null
  */
-function guessBrowser() {
-  if (typeof navigator !== 'undefined' && typeof navigator.userAgent === 'string') {
-    if (/Chrome/.test(navigator.userAgent)) {
-      return 'chrome';
-    }
-    if (/Firefox/.test(navigator.userAgent)) {
-      return 'firefox';
-    }
-    if (/Safari/.test(navigator.userAgent)) {
-      return 'safari';
-    }
+function guessBrowser(userAgent) {
+  if (typeof userAgent === 'undefined') {
+    userAgent = getUserAgent();
+  }
+  if (/Chrome|CriOS/.test(userAgent)) {
+    return 'chrome';
+  }
+  if (/Firefox|FxiOS/.test(userAgent)) {
+    return 'firefox';
+  }
+  if (/Safari/.test(userAgent)) {
+    return 'safari';
   }
   return null;
 }
 
+/**
+ * Guess the browser version.
+ * @param {string} [userAgent=navigator.userAgent]
+ * @returns {?{major: number, minor: number}}
+ */
+function guessBrowserVersion(userAgent) {
+  if (typeof userAgent === 'undefined') {
+    userAgent = getUserAgent();
+  }
+  var prefix = {
+    chrome: 'Chrome|CriOS',
+    firefox: 'Firefox|FxiOS',
+    safari: 'Version'
+  }[guessBrowser(userAgent)];
```

Comment:
This looks bit dangerous as as `guessBrowser` could return null.

Reply:
@makarandp0 `null` results in `prefix` having the value of `undefined`, which is handled in the next line.